Common/GPU/Vulkan/VulkanQueueRunner.cpp

```diff
@@ -376,57 +541,74 @@ void VulkanQueueRunner::PreprocessSteps(std::vector<VKRStep *> &steps) {
 	}
 }
 
-void VulkanQueueRunner::RunSteps(VkCommandBuffer cmd, std::vector<VKRStep *> &steps, QueueProfileContext *profile) {
+void VulkanQueueRunner::RunSteps(FrameData &frameData) {
+	QueueProfileContext *profile = frameData.profilingEnabled_ ? &frameData.profile : nullptr;
+
 	if (profile)
 		profile->cpuStartTime = time_now_d();
 
 	bool emitLabels = vulkan_->Extensions().EXT_debug_utils;
 
-	for (size_t i = 0; i < steps.size(); i++) {
-		const VKRStep &step = *steps[i];
+	for (size_t i = 0; i < frameData.steps.size(); i++) {
+		const VKRStep &step = *frameData.steps[i];
 
 		if (emitLabels) {
 			VkDebugUtilsLabelEXT labelInfo{ VK_STRUCTURE_TYPE_DEBUG_UTILS_LABEL_EXT };
 			labelInfo.pLabelName = step.tag;
-			vkCmdBeginDebugUtilsLabelEXT(cmd, &labelInfo);
+			vkCmdBeginDebugUtilsLabelEXT(frameData.mainCmd, &labelInfo);
 		}
 
 		switch (step.stepType) {
 		case VKRStepType::RENDER:
-			PerformRenderPass(step, cmd);
+			if (!step.render.framebuffer) {
+				_dbg_assert_(!frameData.hasPresentCommands);
```

Comment:
Hm, what about non-buffered?  Is that always only a single render pass?

-[Unknown]